presto-main/src/main/java/com/facebook/presto/operator/scalar/SplitToMapFunction.java

```diff
@@ -109,4 +114,52 @@ public Block splitToMap(@TypeParameter("map<varchar,varchar>") Type mapType, @Sq
 
         return (Block) mapType.getObject(blockBuilder, blockBuilder.getPositionCount() - 1);
     }
+
+    @Description("creates a map using entryDelimiter and keyValueDelimiter")
+    @ScalarFunction("split_to_map")
+    public static class FailOnDuplicateKeys
+    {
+        private final PageBuilder pageBuilder;
+
+        public FailOnDuplicateKeys(@TypeParameter("map<varchar,varchar>") Type mapType)
+        {
+            pageBuilder = new PageBuilder(ImmutableList.of(mapType));
+        }
+
+        @SqlType("map<varchar,varchar>")
+        public Block split(@TypeParameter("map<varchar,varchar>") Type mapType, @SqlType(StandardTypes.VARCHAR) Slice string, @SqlType(StandardTypes.VARCHAR) Slice entryDelimiter, @SqlType(StandardTypes.VARCHAR) Slice keyValueDelimiter)
```

Comment:
nit: this line is too long. Please use https://github.com/airlift/codestyle if you aren't already using it.